test/e2e/upgrade/adminack/adminack.go

```diff
@@ -0,0 +1,134 @@
+package adminack
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	exutil "github.com/openshift/origin/test/extended/util"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+// UpgradeTest contains artifacts used during test
+type UpgradeTest struct {
+	oc     *exutil.CLI
+	config *restclient.Config
+}
+
+func (UpgradeTest) Name() string { return "check-for-admin-acks" }
+func (UpgradeTest) DisplayName() string {
+	return "[bz-Cluster Version Operator] Verify presence of admin ack gate blocks upgrade until acknowledged"
+}
+
+// Setup creates artifacts to be used by Test
+func (t *UpgradeTest) Setup(f *framework.Framework) {
+	g.By("Setting up admin ack test")
+	oc := exutil.NewCLIWithFramework(f)
+	t.oc = oc
+	config, err := framework.LoadConfig()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	t.config = config
+	framework.Logf("Admin ack test setup complete")
+}
+
+// Test simply returns successfully if admin ack functionality is not part the baseline being tested. Otherwise,
+// test first verifies that Upgradeable condition is false for correct reason and with correct message. It then
+// modifies the admin-acks configmap to ack the necessary admin-ack gate and then waits for the Upgradeable
+// condition to change to true.
+func (t *UpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade upgrades.UpgradeType) {
+	// Check if this release has admin ack functionality.
+	gateCm, err := t.oc.AdminKubeClient().CoreV1().ConfigMaps("openshift-config-managed").Get(context.TODO(), "admin-gates", metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			framework.Failf(fmt.Sprintf("Error accessing configmap openshift-config-managed/admin-gates, err=%v", err))
+		} else {
+			framework.Logf("Skipping admin ack test. Admin ack is not in this baseline.")
+			return
+		}
+	}
+	var msg string
+	if msg = gateCm.Data["ack-4.8-kube-122-api-removals-in-4.9"]; msg == "" {
```

Comment:
Sound like the cm name is changed to "ack-4.8-kube-1.22-api-removals-in-4.9" mentioned in https://github.com/openshift/cluster-version-operator/pull/647/files

Reply:
Yes, that was a late breaking change. Thanks for catching.

Reply:
fixed